src/client/datascience/notebookStorage/notebookModelSynchronization.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { traceInfo } from '../../common/logger';
+import { IFileSystem } from '../../common/platform/types';
+import { INotebookEditor, INotebookEditorProvider, INotebookModel, INotebookModelSynchronization } from '../types';
+
+@injectable()
+export class NotebookModelSynchronization implements INotebookModelSynchronization {
```

Comment:
Created this extra class so that NotebookStorageProvider wouldn't have to know about the UI